### PR TITLE
Update rest of the proto files 

### PR DESCRIPTION
**Pull request by Linchin**
**Description of your changes:**
* Update pipeline_spec.proto
* Update job.proto
* Update experiment.proto
* Remove parameter.proto
* Remove resource_reference.proto

This is updated from #6.

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: backend/api/v2beta1/job.proto [hunk 1]
```diff
@@ -61,10 +61,18 @@ option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
...
+  // Creates a new job in an experiment, given the experiment ID.
+  rpc CreateJobByExperiment(CreateJobByExperimentRequest) returns (Job) {
     option (google.api.http) = {
-      post: "/apis/v2beta1/jobs"
+      post: "/apis/v2beta1/experiments/{id}/jobs"
+      body: "job"
+    };
+  }
+
+  // Creates a new job in an experiment, given the experiment ID.
```
Comment (difince) [inline backend/api/v2beta1/job.proto:72]: nit: `given the experiment ID` replace with `given the experiment Name`

File: backend/api/v2beta1/job.proto [hunk 2]
```diff
@@ -130,10 +156,31 @@ message ListJobsRequest {
   // Ascending by default.
   string sort_by = 3;
 
-  // What resource reference to filter on.
-  // E.g. If listing job for an experiment, the query string would be
-  // resource_reference_key.type=EXPERIMENT&resource_reference_key.id=123
-  ResourceKey resource_reference_key = 4;
+  // Optional input. The namespace the jobs belong to. 
+  string namespace = 4;
+
```
Comment (difince) [inline backend/api/v2beta1/job.proto:161]: nit: remove a few empty lines

File: backend/api/v2beta1/job.proto [hunk 3]
```diff
@@ -61,10 +61,18 @@ option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
 
 
 service JobService {
-  // Creates a new job.
-  rpc CreateJob(CreateJobRequest) returns (Job) {
+  // Creates a new job in an experiment, given the experiment ID.
+  rpc CreateJobByExperiment(CreateJobByExperimentRequest) returns (Job) {
     option (google.api.http) = {
-      post: "/apis/v2beta1/jobs"
+      post: "/apis/v2beta1/experiments/{id}/jobs"
```
Comment (difince) [inline backend/api/v2beta1/job.proto:67]: `{id}` needs to comply with a field name in `CreateJobByExperimentRequest`

So, `{id}` needs to become `experiment_id` or via versa

File: backend/api/v2beta1/job.proto [hunk 4]
```diff
@@ -61,10 +61,18 @@ option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
...
+      body: "job"
+    };
+  }
+
+  // Creates a new job in an experiment, given the experiment ID.
+  rpc CreateJobByExperimentName(CreateJobByExperimentNameRequest) returns (Job) {
+    option (google.api.http) = {
+      post: "/apis/v2beta1/experiments/names/{name}/jobs"
```
Comment (difince) [inline backend/api/v2beta1/job.proto:75]: `{name}` needs to comply with a field name in CreateJobByExperimentNameRequest

So, `{name}` needs to become `experiment_name` or via versa

**Review by difince — CHANGES_REQUESTED** [review]
`resource_reference.proto` is still in use in other proto files. So I suggest the removal of the `resourcee_reference.proto` file to happen once all other proto files that depend on it are updated. So, WDYT to still keep resource_reference.proto file in this PR?

File: backend/api/v2beta1/job.proto [hunk 5]
```diff
@@ -130,10 +156,31 @@ message ListJobsRequest {
   // Ascending by default.
   string sort_by = 3;
 
-  // What resource reference to filter on.
-  // E.g. If listing job for an experiment, the query string would be
-  // resource_reference_key.type=EXPERIMENT&resource_reference_key.id=123
-  ResourceKey resource_reference_key = 4;
+  // Optional input. The namespace the jobs belong to. 
+  string namespace = 4;
```
Comment (difince) [inline backend/api/v2beta1/job.proto:160]: `rpc ListJobs` in v1 covers/supports multiple `resource_types`. So if we want to stay compliant with v1 functionality we need to cover all of them. Please, refer to the [document](https://docs.google.com/document/d/1ni1hFPgJRMu834oCuihO1R8TxybkMUeSmxbKfZpbuN0/edit#bookmark=id.enhw8uaef8xg) created by @annajung 
You already have added support for `namespace` and `experiment_id` ( by the new rpc  `ListJobsByExperiment`) .
So in my opinion we need to add least support for `version_id` by adding 

>   // Optional input. The version_id the jobs relate to. 
>   string version_id = 5;

WDYT?

File: backend/api/v2beta1/pipeline_spec.proto [hunk 6]
```diff
@@ -35,11 +34,6 @@ message PipelineSpec {
   // Optional input field. The raw pipeline JSON spec.
```
Comment (difince) [inline backend/api/v2beta1/pipeline_spec.proto:34]: should we remove `string workflow_manifest = 2;` ?

File: backend/api/v2beta1/job.proto [hunk 7]
```diff
@@ -61,10 +61,18 @@ option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
...
+  rpc CreateJobByExperiment(CreateJobByExperimentRequest) returns (Job) {
     option (google.api.http) = {
-      post: "/apis/v2beta1/jobs"
+      post: "/apis/v2beta1/experiments/{id}/jobs"
+      body: "job"
+    };
+  }
+
+  // Creates a new job in an experiment, given the experiment ID.
+  rpc CreateJobByExperimentName(CreateJobByExperimentNameRequest) returns (Job) {
```
Comment (difince) [inline backend/api/v2beta1/job.proto:73]: Now we cover createJobByExperiment (ID or name) ... Should we cover createJob .. by VersionId in order to be compliant with v1?
Again see @annajung's  [doc](https://docs.google.com/document/d/1ni1hFPgJRMu834oCuihO1R8TxybkMUeSmxbKfZpbuN0/edit#bookmark=id.1cbhmh3iipsp)  and take a look in  [getMamifestsBytes](https://github.com/kubeflow/pipelines/blob/7ce1d3750def04ce55a050c50268b8c220c75666/backend/src/apiserver/resource/resource_manager.go#L691-L697) function and how `resource_References` are used. From what I see we need a case that supports `versionId` 
WDYT ?

**Review by difince — CHANGES_REQUESTED** [review]

**Comment by annajung** [comment]
Not all comments from https://github.com/difince/pipelines/pull/6 were addressed in this change, could you take a look at the reviews in the previous PR and apply them here as well? thanks!

File: backend/api/v2beta1/pipeline_spec.proto (hunk 6 above)
Comment (Linchin) [reply backend/api/v2beta1/pipeline_spec.proto:34]: Yes. I will remove it.

**Comment by annajung** [comment]
Hey @Linchin do you have time to look at this? If not, Diana and I are happy to take over and work on the rest of the changes to move this forward. We hope to start moving this forward again

**Pull request closed without merging**